feat(shada): shada should not store nobuflisted buffers (#21818)
neovim/neovim: runtime/doc/options.txt

> 	%	When included, save and restore the buffer list.  If Vim is

Reviewer: Testing `:set shada+=%`, it looks like it is _already_ the case that `'nobuflisted'` buffers are _not_ saved/restored (check with `:ls!`). So the change in this PR actually just fixes a bug, maybe? 

Anyway, let's update the doc here:
```suggestion
	%	Save and restore the buffer list (only 'buflisted' buffers).  If Vim is
```

		started with a file name argument, the buffer list is not
		restored.  If Vim is started without a file name argument, the
		buffer list is restored from the shada file. Quickfix ('buftype'), 

> 		unnamed and buffers on removable media (|shada-r|) are not saved.

Author: Removed `unlisted ('buflisted')` from `shada-%`, as `nobuflisted` files are not stored in general.

Reviewer: Still worth mentioning explicitly I think. Let's keep the help as it was here.

Author: Okay, lets do that. 

The change is reverted by: e88fd1cb512d525d83dde3a1113335e9785f777d
